crates/astria-sequencer/src/app.rs

```diff
@@ -993,6 +1117,38 @@ impl App {
     }
 }
 
+// updates the priority of the txs in the mempool based on the current state,
+// and removes any txs that are now invalid.
+async fn update_mempool_after_finalization<S: StateReadExt>(
+    mempool: &mut Mempool,
+    state: S,
+) -> anyhow::Result<()> {
+    use crate::mempool::TransactionPriority;
+
+    let mut txs_to_remove = Vec::new();
+
+    for (tx, priority) in mempool.inner().await.iter_mut() {
+        let Ok(new_priority) = TransactionPriority::new(
+            tx.nonce(),
+            state
+                .get_account_nonce(Address::from_verification_key(tx.verification_key()))
+                .await
+                .context("failed to fetch account nonce")?,
```

Comment:
failing to get nonce would mean we failed to read from the db, which means any db actions after this will also fail, so it's okay to return an error here imo